### ISSUE TITLE
Mouse Coordinates > 223 don't work

### ISSUE THREAD

**Issue by Sebi-M** [urwid/urwid#476]
##### Description:
Mouse Coordinates > 223 do not work; I cannot drag, drop or click on anything that is beyond that boundary.
The problem lies in the way the mouse coordinates are sent to the application (basically one byte, 256 possible values, excluding additional values for printable characters.)
I already prepared a patch for it; it's a classic "works for me, please review, test further and correct if necessary." I'm no expert in that field.
My version is 2.0.1 as in Debian 10 Buster. The relevant file is unchanged for more than two years, so everything is affected.

Helpful links were:
* https://stackoverflow.com/questions/5966903/how-to-get-mousemove-and-mouseclick-in-bash
* http://invisible-island.net/xterm/ctlseqs/ctlseqs.pdf

##### Affected versions (if applicable)
- [X] `master` branch (specify commit)
- [X] Latest stable version from `pypi`
- [X] Other (specify source)

##### Steps to reproduce (if applicable)
Basically use the script "Global Input" from the tutorials, run it in a terminal with > 223 columns and click (or drag-and-drop) in the rightmost area. There are no events with x-coords > 223.

##### Expected/actual outcome
There should be events arriving.

##### Addendum
The mouse button returned previously was a float sometimes; this was an one-line-fix and submitted as well.

[terminal.patch.txt](https://github.com/urwid/urwid/files/6410533/terminal.patch.txt)
[float.patch.txt](https://github.com/urwid/urwid/files/6410534/float.patch.txt)

**Comment by Sebi-M** [comment]
I made a small mistake regarding the buttons. The line should look like:
`button = ((int(b) & 64) // 64 * 3) + (int(b) & 3) + 1`
Updated patch attached: [terminal.patch.txt](https://github.com/urwid/urwid/files/6414563/terminal.patch.txt)

**Comment by tonycpsu** [comment]
Nice work!  I got a malformed patch error when applying, but manually copy/pasting the code in works, and it does seem to address the mouse problem.  Next time feel free to submit a pull request to make things easier on our end, but happy to have the fix to a long-standing problem that's becoming more relevant nowadays as people work with larger / higher-res screens. :)
